python/src/aiconfig/default_parsers/openai.py

```diff
@@ -505,27 +529,48 @@ def add_prompt_as_message(
         if output.output_type == "execute_result":
             assert isinstance(output, ExecuteResult)
             output_data = output.data
-            # TODO (rossdanlm): Support function call handling as output_data
-            if isinstance(output_data, str):
-                if output.metadata["rawResponse"].get("role", "") == "assistant":
-                    output_message = {
-                        "content": output_data,
-                        "role": "assistant",
-                    }
-                    function_call = output.metadata["rawResponse"].get("function_call", None)
-                    tool_calls = output.metadata["rawResponse"].get("tool_calls", None)
-                    if function_call is not None:
-                        output_message["function_call"] = function_call
-                    if tool_calls is not None:
-                        output_message["tool_calls"] = tool_calls
-                    messages.append(output_message)
+            role = output.metadata.get("assistant", None) or \
+                ("rawResponse" in output.metadata and
+                    output.metadata["rawResponse"].get("assistant", None))
+
+            if role == "assistant":
+                output_message = {}
+                content : Union[str, None] = None
+                function_call : Union[FunctionCallData, None] = None
+                if isinstance(output_data, str):
+                    content = output_data
+                elif isinstance(output_data, OutputDataWithValue):
+                    if isinstance(output_data.value, str):
+                        content = output_data.value
+                    elif output_data.kind == "tool_calls":
+                        assert isinstance(output, OutputDataWithToolCallsValue)
```

Comment:
I feel like this can be simplified:

```
if isinstance(output_data, str):
  # Do string stuff
elif isinstance(output_data, OutputDataWithToolCallsValue):
  # Do function call stuff
elif isinstance(output_data, OutputDataWithVale):
  # Do other value stuff where data.value is string
elif isinstance(output_data, ChatCompletionMessage):
  # Do chat completion stuff

Reply:
I find this more complicated and unintuitive. `OutputDataWithToolCallsValue` is a subset of `OutputDataWithValue`. I get what you're saying that you can have single else-if instead of nested if-statements, but it's jarring to go from "small subset" --> "bigger subset" while reading down through if-else logic. 

This is also more confusing because the inner `# Do string stuff`, `# Do function call stuff` etc all need to individually check for the role type being assistant before we can do anything